stix_shifter_modules/nozomi_vantage/stix_transmission/connector.py

```diff
@@ -0,0 +1,272 @@
+from stix_shifter_utils.modules.base.stix_transmission.base_json_sync_connector import BaseJsonSyncConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+from .api_client import APIClient
+import json
+
+
+class InvalidMetadataException(Exception):
+    pass
+
+
+class Connector(BaseJsonSyncConnector):
+    NOZOMI_MAX_PAGE_SIZE = 1000
+
+    def __init__(self, connection, configuration):
+        self.api_client = APIClient(connection, configuration)
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+        self.expired_token_count = 0
+
+    async def create_results_connection(self, query, offset, length, metadata=None):
+        """
+        Fetching the results using query, offset and length
+        :param query: str, Data Source query
+        :param offset: str, Offset value
+        :param length: str, Length value
+        :param metadata: dict
+        :return: return_obj, dict
+        """
+        data = []
+        try:
+            result_count, total_records, page_number, page_size = self.handle_metadata_and_page_size(offset, length,
+                                                                                                     metadata)
+            # Generating jwt token
+            jwt_token, return_obj = await self.__get_token()
+            if return_obj:
+                return return_obj
+
+            if (result_count == 0 and page_number == 1) or \
+                    (page_number and result_count < self.api_client.result_limit):
+
+                while result_count < total_records:
+                    query_url = f'{query}&page={page_number}&count={page_size}'
+                    response_wrapper = await self.api_client.get_search_results(query_url, jwt_token)
+                    response_dict, return_obj = self.handle_api_response(response_wrapper)
+
+                    if return_obj:
+                        return return_obj
+
+                    # Generate a new token only the token_generated_count is 1.
+                    # If it is more than one, it has already been generated.
+                    if self.expired_token_count == 1:
+                        jwt_token, return_obj = await self.__get_token()
+                        self.expired_token_count += 1
+                        if return_obj:
+                            return return_obj
+                        continue
+
+                    return_obj['success'] = True
+                    data += response_dict['result']
+                    result_count += len(response_dict['result'])
+
+                    if response_dict['total'] <= result_count or len(data) == 0:
```

Comment:
This is probably ok, but I think checking if your 
response_dict['result'] < page_size
is a better approach. If your current api results are less than the page_size you asked for, you've gotten all of the data and can exit the loop.

The logic is a bit easier to follow imo.

Reply:
Updated as per the suggestion.